### PR TITLE
Move part of ODFE fields into DKAN Dataset

### PR DESCRIPTION
**Pull request by ghost**
Ref: https://github.com/NuCivic/dkan/pull/1301

NOTE: Please revert the 'open_data_schema_map' and 'open_data_schema_map_dkan' branches before merging this PR. Check notes on references PR.

### CODE REVIEW

File: dkan_dataset.forms.inc [hunk 1]
```diff
@@ -187,6 +201,46 @@ function dkan_dataset_form_alter(&$form, &$form_state, $form_id) {
...
+      // Groups.
+      $og_group_ref_langcode = dkan_dataset_form_field_language($form, 'og_group_ref');
+      $form['og_group_ref'][$og_group_ref_langcode][0]['default']['#required'] = TRUE;
+      // Check if there are any groups. At least there is one option which is the '-None-' option.
+      // If there are no groups then display a warning letting the user know that he won't be able
+      // to create the dataset since the 'Publisher' field is a POD required field.
+      $available_groups = $form['og_group_ref'][$og_group_ref_langcode][0]['default'];
+      if (isset($available_groups['#options']) && count($available_groups['#options']) <= 1) {
+        // Show a warning message.
+        display_groups_warning();
```
Comment (dafeder) [inline dkan_dataset.forms.inc]: Namespace this function!

File: dkan_dataset.forms.inc [hunk 2]
```diff
@@ -380,6 +434,35 @@ function dkan_dataset_dataset_form_submit($form, &$form_state) {
...
+  // is enabled in DKAN Dataset Forms.
+  // The check about the field being required and empty won't work because the 'Groups'
+  // field is hidden when the user is not associated with any groups.
+  if (variable_get('dkan_dataset_form_group_validation') && module_exists('dkan_dataset_groups')) {
+    // Get available groups from the field options.
+    $field_og_group_ref_langcode = dkan_dataset_form_field_language($form, 'og_group_ref');
+    $available_groups = $form['og_group_ref'][$field_og_group_ref_langcode][0]['default'];
+    // If there are no options or only one (-None-) then display an error.
+    if (isset($available_groups['#options']) && count($available_groups['#options']) <= 1) {
+      display_groups_warning();
```
Comment (dafeder) [inline dkan_dataset.forms.inc]: Namespace

File: dkan_dataset.forms.inc [hunk 3]
```diff
@@ -380,6 +434,35 @@ function dkan_dataset_dataset_form_submit($form, &$form_state) {
...
+      form_set_error('', 'Error: You tried to submit a dataset with no groups assigned.');
+    }
+  }
+}
+
+/**
+ * Display groups warning on dataset form.
+ */
+function display_groups_warning() {
```
Comment (dafeder) [inline dkan_dataset.forms.inc]: should be dkan_dataset_groups_warning() or something

File: dkan_dataset.forms.inc [hunk 4]
```diff
@@ -187,6 +201,46 @@ function dkan_dataset_form_alter(&$form, &$form_state, $form_id) {
...
+    // Mark 'Groups' field as required if the group validation is enabled on DKAN Dataset forms.
+    if (variable_get('dkan_dataset_form_group_validation') && module_exists('dkan_dataset_groups')) {
+      // Groups.
+      $og_group_ref_langcode = dkan_dataset_form_field_language($form, 'og_group_ref');
+      $form['og_group_ref'][$og_group_ref_langcode][0]['default']['#required'] = TRUE;
+      // Check if there are any groups. At least there is one option which is the '-None-' option.
+      // If there are no groups then display a warning letting the user know that he won't be able
+      // to create the dataset since the 'Publisher' field is a POD required field.
+      $available_groups = $form['og_group_ref'][$og_group_ref_langcode][0]['default'];
+      if (isset($available_groups['#options']) && count($available_groups['#options']) <= 1) {
```
Comment (ghost) [inline dkan_dataset.forms.inc]: Don't show warning if groups field is unhidden

File: dkan_dataset.forms.inc [hunk 5]
```diff
@@ -380,6 +434,35 @@ function dkan_dataset_dataset_form_submit($form, &$form_state) {
...
 
 /**
+ * Validation function for dataset node form.
+ */
+function dkan_dataset_dataset_node_form_validate($form, &$form_state) {
+  // Throw an error if the user has no groups available but the strict POD validation
+  // is enabled in DKAN Dataset Forms.
+  // The check about the field being required and empty won't work because the 'Groups'
+  // field is hidden when the user is not associated with any groups.
+  if (variable_get('dkan_dataset_form_group_validation') && module_exists('dkan_dataset_groups')) {
```
Comment (ghost) [inline dkan_dataset.forms.inc:459]: Don't show warning if groups field is unhidden

File: modules/dkan_dataset_content_types/dkan_dataset_content_types.features.conditional_fields.inc [hunk 6]
```diff
@@ -0,0 +1,225 @@
...
+            3 => 0,
+            4 => 0,
+          ),
+        ),
+        'element_edit' => array(
+          1 => 1,
+          3 => 0,
+        ),
+        'element_edit_per_role' => 0,
+        'element_edit_roles' => array(
```
Comment (ghost) [inline modules/dkan_dataset_content_types/dkan_dataset_content_types.features.conditional_fields.inc]: Check if this role export is necessary.

**Pull request merged**